lib/railway_ipc/rabbit_mq/rabbit_mq_adapter.ex

```diff
@@ -68,10 +72,30 @@ defmodule RailwayIpc.RabbitMQ.RabbitMQAdapter do
     Queue.declare(channel, queue, opts)
   end
 
+  def maybe_create_exchange(_channel, nil) do
+    :ok
+  end
+
+  def maybe_create_exchange(channel, exchange) do
+    Exchange.declare(channel, exchange, :fanout, durable: true)
+  end
+
+  def maybe_bind_queue(_channel, _queue, nil) do
+    :ok
+  end
+
+  def maybe_bind_queue(channel, queue, exchange) do
+    Queue.bind(channel, queue, exchange)
+  end
+
   def ack(channel, delivery_tag) do
     Basic.ack(channel, delivery_tag)
   end
 
+  def publish(channel, nil, queue, payload) do
```

Comment:
This is for the direct publishing to the re-published messages queue. We could re-use this function later when we need to directly publish messages that need to be re-consumed. Another option is to be more explicit with this code flow and call this something like `direct_publish` and call that where appropriate.